forte/processors/base/base_processor.py

```diff
@@ -74,7 +101,8 @@ def check_record(self, input_pack: PackType):
         """
         if self._check_type_consistency:
             expectation = self.expected_types_and_attributes()
-            record_types_and_attributes_check(expectation, input_pack)
+            input_pack_record = self.collect_input_pack_record(input_pack)
+            record_types_and_attributes_check(expectation, input_pack_record)
```

Comment:
I think you can simply do it in line.

```suggestion
            record_types_and_attributes_check(self.expected_types_and_attributes(), self.collect_input_pack_record(input_pack))
```